client/my-sites/site-settings/settings-performance/main.jsx

```diff
@@ -32,122 +32,149 @@ import isPrivateSite from 'calypso/state/selectors/is-private-site';
 import { getSelectedSite, getSelectedSiteId } from 'calypso/state/ui/selectors';
 import { getSiteSlug, isJetpackSite } from 'calypso/state/sites/selectors';
 import config from '@automattic/calypso-config';
+import { useExperiment } from 'calypso/lib/explat';
+import { getSitePurchases } from 'calypso/state/purchases/selectors';
+import { isJetpackSearch, isP2Plus, planHasJetpackSearch } from '@automattic/calypso-products';
+
+function SiteSettingsPerformance( props ) {
+	const {
+		fields,
+		handleAutosavingToggle,
+		hasSearchProduct,
+		isRequestingSettings,
+		isSavingSettings,
+		onChangeField,
+		saveJetpackSettings,
+		showCloudflare,
+		site,
+		siteId,
+		siteIsAtomic,
+		siteIsAtomicPrivate,
+		siteIsJetpack,
+		siteIsUnlaunched,
+		siteSlug,
+		submitForm,
+		trackEvent,
+		translate,
+		updateFields,
+	} = props;
+	const siteIsJetpackNonAtomic = siteIsJetpack && ! siteIsAtomic;
+
+	const { isLoading, experimentAssignment } = useExperiment(
+		'jetpack_search_performance_settings_placement',
+		// Targets experiment assignments to users without search purchases but eligible for Cloudflare promo.
+		{ isEligible: showCloudflare && ! siteIsJetpackNonAtomic && ! hasSearchProduct }
```

Comment:
It looks like the eligibility here is different to below where we choose what order to display the Cloudflare component. Once a user `hasSearchProduct`, it is possible the order might flip back, is that alright?

Reply:
Ah, so the experiments hook defaults to the control variation when it's no longer eligible. (It's so obvious when I think about it 😆)

Yes, it'll be fine if the order flips back following a search product purchase/acquisition.